editor/src/components/canvas/controls/breadcrumb-trail.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react'
+import { useEditorState } from '../../../components/editor/store/store-hook'
+import { ElementPathElement } from '../../../components/inspector/sections/header-section/element-path'
+import {
+  betterReactMemo,
+  useKeepReferenceEqualityIfPossible,
+} from '../../../utils/react-performance'
+import { Utils } from '../../../uuiui-deps'
+import * as TP from '../../../core/shared/template-path'
+import { getOpenUtopiaJSXComponentsFromState } from '../../editor/store/editor-state'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { selectComponents } from '../../../components/editor/actions/action-creators'
+import { FlexRow, Icons } from '../../../uuiui'
+
+export const BreadcrumbTrail = () => {
+  const { dispatch, jsxMetadataKILLME, rootComponents } = useEditorState((store) => {
+    return {
+      dispatch: store.dispatch,
+      jsxMetadataKILLME: store.editor.jsxMetadataKILLME,
+      rootComponents: getOpenUtopiaJSXComponentsFromState(store.editor),
+    }
+  }, 'InspectorContextProvider')
+
+  const onSelect = React.useCallback(
+    (path) => dispatch([selectComponents([path], false)], 'everyone'),
+    [dispatch],
+  )
+
+  const selectedViews = useEditorState(
+    (store) => store.editor.selectedViews,
+    'InspectorEntryPoint selectedViews',
+  )
+
+  const elementPath = useKeepReferenceEqualityIfPossible(
+    React.useMemo(() => {
+      if (selectedViews.length === 0) {
+        return []
+      }
+      let elements: Array<ElementPathElement> = []
+      Utils.fastForEach(TP.allPaths(selectedViews[0]), (path) => {
+        // TODO Scene Implementation
+        if (TP.isInstancePath(path)) {
+          const component = MetadataUtils.getElementByInstancePathMaybe(
+            jsxMetadataKILLME.elements,
+            path,
+          )
+          if (component != null) {
+            elements.push({
+              name: MetadataUtils.getElementLabel(path, jsxMetadataKILLME),
+              path: path,
+            })
+          }
+        } else {
+          const scene = MetadataUtils.findSceneByTemplatePath(jsxMetadataKILLME.components, path)
+          if (scene != null) {
+            elements.push({
+              name: scene.label,
+              path: path,
+            })
+          }
+        }
+      })
+      return elements
+    }, [selectedViews, jsxMetadataKILLME]),
+  )
+
+  const ElementPathButtons = betterReactMemo('ElementPathButtons', () => {
```

Comment:
Please pull this component up to the top level as otherwise we're creating inner components here that will cause issues with performance and memoisation.

Reply:
since the main component seems to be only rendering this, I think it's even easier to just "inline" this component. ie just render it without making it a component first.

Reply:
This was legacy inspector code that I just asked Omar to copy over. Good occasion to clean it up!

Reply:
Commit #930332e address this